conditions/issuer.go

```diff
@@ -17,59 +17,65 @@ limitations under the License.
 package conditions
 
 import (
-	cmutil "github.com/cert-manager/cert-manager/pkg/api/util"
 	cmapi "github.com/cert-manager/cert-manager/pkg/apis/certmanager/v1"
 	cmmeta "github.com/cert-manager/cert-manager/pkg/apis/meta/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/utils/clock"
 )
 
-// private struct; only used to implement the GenericIssuer interface
-// for use with the cmutil.SetIssuerCondition function
-type genericIssuer struct {
-	metav1.TypeMeta   `json:",inline"`
-	metav1.ObjectMeta `json:"metadata,omitempty"`
-
-	Status cmapi.IssuerStatus `json:"status"`
-}
-
-func (g *genericIssuer) DeepCopyObject() runtime.Object {
-	panic("[HACK]: this function should not get called")
-}
-
-func (g *genericIssuer) GetSpec() *cmapi.IssuerSpec {
-	panic("[HACK]: this function should not get called")
-}
-
-func (g *genericIssuer) GetObjectMeta() *metav1.ObjectMeta {
-	return &g.ObjectMeta
-}
-
-func (g *genericIssuer) GetStatus() *cmapi.IssuerStatus {
-	return &g.Status
-}
-
 // Update the status with the provided condition details & return
 // the added condition.
-// NOTE: this code is just a workaround for cmutil only accepting a GenericIssuer interface
```

Comment:
Instead of calling the cert-manager code, we now just host this logic in this repo. This makes it much clearer what is going on & allows us to be more efficient.